test/governance/ProxyAdminWithDeputy.test.js

```diff
@@ -0,0 +1,139 @@
+const { expect } = require("chai")
+
+const { upgrades } = require("hardhat")
+
+describe("ProxyAdminWithDeputy", () => {
+  let deployer
+  let deputy
+  let timelock
+  let adminWithDeputy
+
+  let SimpleStorage
+  const initialState = 42
+
+  beforeEach(async () => {
+    ;[deployer, deputy, timelock] = await ethers.getSigners()
+    SimpleStorage = await ethers.getContractFactory("SimpleStorage")
+    const initializerArgs = [initialState] // stored value in proxy state
+    storage = await upgrades.deployProxy(SimpleStorage, initializerArgs, {
+      kind: "transparent",
+      constructorArgs: [1], // implementation version 1
+    })
+    await storage.deployed()
+
+    const GovernorStub = await ethers.getContractFactory(
+      "TestTokenholderGovernorStubV2"
+    )
+    tGov = await GovernorStub.deploy(timelock.address)
+    await tGov.deployed()
+
+    const ProxyAdminWithDeputy = await ethers.getContractFactory(
+      "ProxyAdminWithDeputy"
+    )
+    adminWithDeputy = await ProxyAdminWithDeputy.deploy(
+      tGov.address,
+      deputy.address
+    )
+    await adminWithDeputy.deployed()
+  })
+
+  describe("Plain Upgrades deployment - No ProxyAdminWithDeputy", () => {
+    let newImplementationAddress
+    let adminInstance
+
+    beforeEach(async () => {
+      newImplementationAddress = await upgrades.prepareUpgrade(
+        storage.address,
+        SimpleStorage,
+        {
+          constructorArgs: [2],
+        }
+      )
+      adminInstance = await upgrades.admin.getInstance()
+    })
+
+    it("ProxyAdmin is the admin for the UpgradeableProxy", async () => {
+      const adminInstance = await upgrades.admin.getInstance()
```

Comment:
It's already computed - see the last line of `beforeEach`.

Reply:
✅